erpnext/manufacturing/doctype/job_card/job_card.js

```diff
@@ -39,7 +39,7 @@ frappe.ui.form.on('Job Card', {
 		}
 
 		if (frm.doc.docstatus == 0 && (frm.doc.for_quantity > frm.doc.total_completed_qty || !frm.doc.for_quantity)
-			&& (!frm.doc.items.length || frm.doc.for_quantity == frm.doc.transferred_qty)) {
+			&& (frm.doc.items || !frm.doc.items.length || frm.doc.for_quantity == frm.doc.transferred_qty)) {
```

Comment:
```suggestion
			&& (frm.doc.for_quantity == frm.doc.transferred_qty)) {
```
your condition currently suggests that that condition will pass if "items are there" and even if "theres no length for items", then why do you need item condition if it works for wither of the case?